src/main/java/com/disraely/services/BaseService.java

```diff
@@ -0,0 +1,48 @@
+package com.disraely.services;
+
+import com.disraely.entities.BaseEntity;
+import com.disraely.enums.EntityStatus;
+import com.disraely.interfaces.IBaseService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BaseService<E extends BaseEntity> implements IBaseService<E> {
+
+    protected final List<E> entities;
+    protected final Class<E> entityClass;
+
+    public BaseService(Class<E> entityClass) {
+        this.entities = new ArrayList<E>();
+        this.entityClass = entityClass;
+    }
+
+    @Override
+    public Optional<E> findOneById(long id) {
+        return Optional.empty();
+    }
+
+    @Override
+    public List<E> find() {
+        return this.entities
+                .stream()
+                .filter(e -> e.getStatus() != EntityStatus.DELETED)
+                .toList();
+    }
+
+    @Override
+    public boolean delete(long id) {
```

Comment:
No need to return boolean. You can return nothing or the entity that you deleted